src/Groups/group_constructors.jl

```diff
@@ -107,24 +127,28 @@ function abelian_group(v::Vector{Int})
 end
 =#
 
-"""
+@doc Markdown.doc"""
     abelian_group(::Type{T}, v::Vector{Int}) where T <: Group -> PcGroup
 
-Return the direct product of cyclic groups of order v[1] x v[2] x ... x v[n], as group of type `T`. Here, `T` must be of type `PermGroup`, `FPGroup` or `PcGroup`.
+Return the direct product of cyclic groups of the orders
+`v[1]`, `v[2]`, $\ldots$, `v[n]`, as an instance of `T`.
+Here, `T` must be one of `PermGroup`, `FPGroup`, or `PcGroup`.
+
+!!! warning
+    The type need to be specified in the input of the function `abelian_group`, otherwise a group of type `GrpAbFinGen` is returned, which is not a GAP group type. In future versions of Oscar, this may change.
 """
 function abelian_group(::Type{T}, v::Vector{Int}) where T <: GAPGroup
-  v1 = GAP.julia_to_gap(v)
-  return T(GAP.Globals.AbelianGroup(_gap_filter(T), v1))
+  return T(GAP.Globals.AbelianGroup(_gap_filter(T), GAP.julia_to_gap(v)))
```

Comment:
Assuming it works (I didn't test), generally I find this nicer / more "Julia-ish":
```suggestion
  return T(GAP.Globals.AbelianGroup(_gap_filter(T), GAP.GapObj(v)))
```

Reply:
Yes. (I had already replaced several occurrences.)